File: android/app/src/main/java/my/ssdid/wallet/domain/verifier/offline/CredentialRepository.kt
package my.ssdid.wallet.domain.verifier.offline

import my.ssdid.wallet.domain.model.VerifiableCredential

interface CredentialRepository {
    suspend fun saveCredential(credential: VerifiableCredential)
    suspend fun getHeldCredentials(): List<VerifiableCredential>
    suspend fun getUniqueIssuerDids(): List<String>
    suspend fun deleteCredential(credentialId: String)
}

File: sdk/ios/SsdidCore/Sources/SsdidCore/Domain/Verifier/Offline/CredentialRepository.swift
import Foundation

/// Persistent storage for held (received) verifiable credentials.
public protocol CredentialRepository: Sendable {
    func saveCredential(_ credential: VerifiableCredential) async throws
    func getHeldCredentials() async -> [VerifiableCredential]
    func getUniqueIssuerDids() async -> [String]
    func deleteCredential(credentialId: String) async throws
}
